doc/api/modules.md

```diff
@@ -405,8 +405,13 @@ always return the built in HTTP module, even if there is `require.cache` entry
 by that name.
 
 Some core modules are always preferentially loaded if their identifier is
-passed to `require()`. For instance, `require('http')` will always
-return the built-in HTTP module, even if there is a file by that name. The list
+passed to `require()`. For instance, `require('http')` will 
+return the built-in HTTP module, even if there is a file by that name. However, 
+note that core modules specified without the `node:` prefix are served from the 
+cache if an entry for the module exists. Because cache entries can be manually 
+altered by modifying the `require.cache `object, this allows the module's 
+implementation to be replaced. To ensure `require() ` returns the original 
+implementation, use the `node:` prefix. The list
```

Comment:
As a newcomer to this, I find the wording "assuming no mutation of the module cache" somewhat confusing. 
My goal is to clearly understand the specification, and this phrase doesn't help clarify the situation for me.
I believe my initial wording is preferable because it explicitly states the technical condition under which the behavior differs.

Reply:
How is it confusing, and how can we clear up that confusion?

Reply:
I find the wording "assuming no mutation of the module cache" somewhat confusing because it implies that mutations exist, which begs the question: "under what circumstances?" My wording describes the specification as it is. As for how to move forward, feel free to decide how to write this.

Reply:
I think that's fine, the rest of the documentation should contain the answer to that question. Maybe it's clearer if we say `require.cache` instead of "the module cache"
```suggestion
passed to `require()`. For instance, assuming `require.cache` is not modified, `require('http')`
will return the built-in HTTP module, even if there is a file by that name. The list
```

If you look at the section just above this one, it already deals with how caching works in CJS

Reply:
IMO it should be past tense, 'assuming `require.cache` has not been modified'

Reply:
FYI, I initially raised this question in the IRC, and the reply I got indicated that there is an impression that core modules are not served from the cache at all, which suggests that this is not so clear.  Just my 2c.